mcs_benchmark_data/metadata_template_metadata.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+from dataclasses import dataclass
+from pathlib import Path
+
+
+@dataclass(frozen=True)
+class MetadataTemplateMetadata:
+    """
+    Information about the metadata file necessary to create a file from a template
+    """
+
+    benchmark_name: str
+    submission_name: Optional[str] = None
+    template_name: Optional[str] = None
+    dest_file_name: Optional[str] = None
+    dest_file_path_from_root: Optional[Path] = None
+
+    def __post_init__(self):

```

Comment:
This is hacky. Just do a non-frozen `@dataclass` with `init=True`. See the dataclasses module documentation.